common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1815,31 +1834,45 @@ func (d *cassandraPersistence) UpdateWorkflowExecution(request *p.InternalUpdate
 		d.createTimerTasks(batch, startReq.TimerTasks, startReq.DomainID, startReq.Execution.GetWorkflowId(),
 			startReq.Execution.GetRunId(), cqlNowTimestamp)
 	} else {
-		startVersion := common.EmptyVersion
-		lastWriteVersion := common.EmptyVersion
-		if request.ReplicationState != nil {
-			startVersion = request.ReplicationState.StartVersion
-			lastWriteVersion = request.ReplicationState.LastWriteVersion
+		switch executionInfo.State {
+		case p.WorkflowStateZombie:
```

Comment:
Why this only do a sanity check?

Reply:
i do not think there is a conditional get operation in cassandra
we need to have this check here, at least for the initial testing for 3+DC
for prod run, we can decide later